content/5_Plat/2DRQ.mdx

```diff
@@ -207,11 +208,134 @@ factors for both are terrible.
 
 This doesn't require knowing the queries beforehand.
 
-- [my 1D offline BIT](<https://github.com/bqi343/USACO/blob/master/Implementations/content/data-structures/1D%20Range%20Queries%20(9.2)/BIToff.h>)
-- [my 2D offline BIT](<https://github.com/bqi343/USACO/blob/master/Implementations/content/data-structures/2D%20Range%20Queries%20(15.2)/BIT2DOff%20(15.2).h>)
+- [Benq's 1D offline BIT](<https://github.com/bqi343/USACO/blob/master/Implementations/content/data-structures/1D%20Range%20Queries%20(9.2)/BIToff.h>)
+- [Benq's 2D offline BIT](<https://github.com/bqi343/USACO/blob/master/Implementations/content/data-structures/2D%20Range%20Queries%20(15.2)/BIT2DOff%20(15.2).h>)
 
-It's a bit difficult to pass the above problem within the time limit. Make sure
-to use fast input (and not `endl`)!
+Here's an implementation of the offline 2D BIT presented above that may be
+easier to understand, albeit significantly slower (same complexity, just worse constant factor):
+
+<LanguageSection>
+<CPPSection>
+```cpp
+// see code example below for usage
+struct BIT2D {
+	vector<int> vals[N], bit[N];
```

Comment:
maybe have `N` passed in as constructor?

Reply:
I pass it as a template argument, is that good enough

Reply:
why does it have to be compile-time